Issue with -lyaml library when using 84codes/crystal:latest-alpine

Hello,

I'm currently working on building a Crystal application and switched from `crystallang/crystal:latest-alpine` to `84codes/crystal:latest-alpine` to enable ARM support. However, I encountered an issue during the build process, specifically:


```
#12 91.25 E: Error target noir failed to compile:
#12 91.25 /usr/lib/gcc/aarch64-alpine-linux-musl/14.2.0/../../../../aarch64-alpine-linux-musl/bin/ld: cannot find -lyaml (this usually means you need to install the development package for libyaml): No such file or directory
#12 91.25 collect2: error: ld returned 1 exit status
#12 91.25 Error: execution of command failed with exit status 1: cc "${@}" -o /noir/bin/noir  -rdynamic -static -L/usr/bin/../lib/crystal -lz `command -v pkg-config > /dev/null && pkg-config --libs --silence-errors libssl || printf %s '-lssl -lcrypto'` `command -v pkg-config > /dev/null && pkg-config --libs --silence-errors libcrypto || printf %s '-lcrypto'` -lyaml -lpcre2-8 -lgc
```

![](https://github.com/user-attachments/assets/cd43010c-ece4-4e4d-8f3c-0c4c2aaea11d)
*crystallang image succeeds, 84codes image fails*

```dockerfile
##= BUILDER =##
FROM 84codes/crystal:latest-alpine As builder

WORKDIR /noir
COPY . .

RUN shards install --production && \
    shards build --release --no-debug --production --static
# Ref: https://crystal-lang.org/reference/1.15/guides/static_linking.html
```
*Modify only the FROM part in https://github.com/owasp-noir/noir/blob/a8b0789a67dfade24cdf73d2a8bb0c9abef17281/Dockerfile*

I initially suspected the issue could be related to the yaml-dev, musl-dev packages, and I tried installing it via `apk add`, but the error persists. I also believe the yaml library might already be installed within the image.

It seems like the build is unable to locate the library, which makes me think it might be related to the library path configuration. Could you provide any insights or suggestions on how I might resolve this issue?

Thank you in advance for your help!

--- a/alpine/Dockerfile
+++ b/alpine/Dockerfile
@@ -53,7 +53,7 @@ RUN sh -ex bin/shards.sh && strip bin/shards
 # start from a clean image
 FROM alpine:$alpine_version
 # add dependencies commonly required for building crystal applications
-RUN apk add --update --no-cache musl-dev gcc pcre2-dev libevent-dev libevent-static openssl-dev openssl-libs-static libxml2-dev zlib-dev zlib-static git make yaml-dev libxml2-static gmp-dev xz-static
+RUN apk add --update --no-cache musl-dev gcc pcre2-dev libevent-dev libevent-static openssl-dev openssl-libs-static libxml2-dev zlib-dev zlib-static git make yaml-dev libxml2-static gmp-dev xz-static yaml-static
 # copy the binaries + stdlib + libgc from the build stage
 COPY --from=builder /usr/src/crystal/*.md /usr/share/doc/crystal/
 COPY --from=builder /usr/src/crystal/man/crystal.1.gz /usr/share/man/man1/

